src/main/java/edu/harvard/iq/dataverse/DatasetVersionFilesServiceBean.java

```diff
@@ -230,4 +262,29 @@ private void applyOrderCriteriaToGetFileMetadatasQuery(JPAQuery<FileMetadata> qu
                 break;
         }
     }
+
+    private long getOriginalTabularFilesSize(DatasetVersion datasetVersion) {
+        JPAQueryFactory queryFactory = new JPAQueryFactory(em);
+        Long result = queryFactory
+                .from(fileMetadata)
+                .where(fileMetadata.datasetVersion.id.eq(datasetVersion.getId()))
+                .from(dataTable)
+                .where(fileMetadata.dataFile.dataTables.isNotEmpty().and(dataTable.dataFile.eq(fileMetadata.dataFile)))
```

Comment:
Yes, `.where(dataTable.dataFile.eq(fileMetadata.dataFile))` 
should be sufficient - and it simplifies the resulting query a bit.